mc-java-comparable-tests/src/test/kotlin/io/spine/tools/mc/java/comparable/AddComparatorErrorSpec.kt

```diff
@@ -0,0 +1,251 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.mc.java.comparable
+
+import io.kotest.matchers.string.shouldContain
+import io.spine.tools.mc.java.comparable.given.BytesProhibited
+import io.spine.tools.mc.java.comparable.given.MapsProhibited
+import io.spine.tools.mc.java.comparable.given.NestedBytesProhibited
+import io.spine.tools.mc.java.comparable.given.NestedMapsProhibited
+import io.spine.tools.mc.java.comparable.given.NestedMessage
+import io.spine.tools.mc.java.comparable.given.NestedNonComparableProhibited
+import io.spine.tools.mc.java.comparable.given.NestedNonExistingProhibited
+import io.spine.tools.mc.java.comparable.given.NestedOneOfProhibited
+import io.spine.tools.mc.java.comparable.given.NestedRepeatedProhibited
+import io.spine.tools.mc.java.comparable.given.NoCompareByOption
+import io.spine.tools.mc.java.comparable.given.NonComparableProhibited
+import io.spine.tools.mc.java.comparable.given.NonExistingProhibited
+import io.spine.tools.mc.java.comparable.given.OneOfProhibited
+import io.spine.tools.mc.java.comparable.given.RepeatedProhibited
+import java.nio.file.Path
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+
+@DisplayName("`AddComparator` should report an error if '(compare_by)' option refers to")
+internal class AddComparatorErrorSpec {
+
+    /**
+     * The piece of the error message which lists the supported types.
+     */
+    private val supportedTypes =
+        "Supported field types are: primitives, enums, and comparable messages."
+
+    /**
+     * The part of the error message instructing the user to read the docs.
+     */
+    private val callToAction = "Please see the `(compare_by)` option documentation for details."
+
+    /**
+     * The introduction part of the error message telling that maps and repeated fields are
+     * not supported for comparison.
+     */
+    private val repeatedAndMapsProlog = "Repeated fields or maps cannot participate in comparison."
+
+    private lateinit var projectDir: Path
+
+    @BeforeEach
+    fun createProjectDir(@TempDir dir: Path) {
+        projectDir = dir
+    }
+
+    @Test
+    fun `a non-comparable message`() {
+        val error = compile<NonComparableProhibited>(projectDir)
+        error.message.let {
+            // The field path.
+            it shouldContain "`id`"
+            // The type of the message declaring the field.
+            it shouldContain NonComparableProhibited.getDescriptor().name
+            // The type of the field.
+            it shouldContain NoCompareByOption.getDescriptor().name
+            // The call to action.
+            it shouldContain callToAction
+        }
+    }
+
+    @Test
+    fun `a 'bytes' field`() {
```

Comment:
```suggestion
    fun `a bytes field`() {
```

These quoting is a bit inconsistent within the file.